ext/standard/string.c

```diff
@@ -4006,7 +4007,9 @@ static zend_long php_str_replace_in_subject(zval *search, zval *replace, zval *s
 				/* Get current entry */
 				if ((replace_entry = zend_hash_get_current_data_ex(Z_ARRVAL_P(replace), &pos)) != NULL) {
 					/* Make sure we're dealing with strings. */
-					convert_to_string_ex(replace_entry);
+					ZVAL_DEREF(replace_entry);  // Deal with possibly shared zvals
```

Comment:
zval_get_string segfaults if done before [L402](https://github.com/php/php-src/pull/1022/files#diff-b31234a9f5a03a328b60d0453988140fR4012) (convert_to_string), don't know for sure why.

I'll try that again, anyway.